### PR TITLE
Test suite enhancements around WMCO deployments

### PR DESCRIPTION
**Pull request by sebsoto**
This is a set of changes which are meant to enable better upgrade testing in CI

### CODE REVIEW

File: test/e2e/validation_test.go [hunk 1]
```diff
@@ -41,6 +42,9 @@ const (
 	wmcoContainerName = "manager"
 )
 
+// versionRegex captures the version from the output of the WMCO version command
+var versionRegex = regexp.MustCompile(`version: "([^"]*)"`)
```
Comment (aravindhp) [inline test/e2e/validation_test.go:46]: Please add an example to show what you are matching for

**Review by aravindhp — APPROVED** [review]
Thanks for doing this, @sebsoto. No more disabling the version test when running locally ❤️

**Review by saifshaikh48 — COMMENTED** [review]
/lgtm

**Comment by sebsoto** [comment]
/test aws-e2e-upgrade

Some issues installing the operator, investigating.

**Comment by aravindhp** [comment]
/hold 
until kube 1.27 rebase https://github.com/openshift/windows-machine-config-operator/pull/1601 lands

**Pull request merged**